token/program-2022/src/processor.rs

```diff
@@ -784,35 +784,40 @@ impl Processor {
             }
         }
 
-        match source_account.base.delegate {
-            COption::Some(ref delegate) if cmp_pubkeys(authority_info.key, delegate) => {
-                Self::validate_owner(
+        if !source_account
+            .base
+            .is_owned_by_system_program_or_incinerator()
+        {
+            match source_account.base.delegate {
```

Comment:
I guess dealing with the case of a token owned by the incinerator but with an active delegate is too pedantic, wfm :)